docs/proposals/update-downloader.md

```diff
@@ -112,55 +108,50 @@ timeouts). Also we miss optional requirement 1.2.
   happening. Also it can happen that the user accidentally modifies
   the update file (also unlikely, but possible).
 
-In terms of general design, these two drawbacks probably stem from two
+In terms of general design, these two drawbacks stem from two
 more general problems:
-1. We basically assume that writing installer on disk and updating
-   `MVar` is a single action which happens atomically. But we don't
+1. We basically assume that writing the installer on disk and updating
+   `MVar` is a single action which happens atomically. But we do not
    have any guarantees about correspondence between the file on disk
    and the contents of `MVar`.
-2. When we have an installer on disk, we don't know which installer
-   it's supposed to be. We just assume that it's the one we want to
+2. When we have an installer on disk, we do not know which installer
+   it is supposed to be. We just assume that it is the one we want to
    install.
 
 ## Suggested improvements
 
 To address the drawbacks of the current approach, I propose to do the
 following changes:
-1. Change the check whether we have already downloaded the installer
-   in `downloadUpdateDo` function. If the file doesn't exist, we can
-   simply proceed with download. If the file exists, there are two
+1. Change the already downloaded check for the installer
+   in the `downloadUpdateDo` function. If the file does not exist, proceed with the download. If the file exists, there are two
    cases:
    * It contains the installer we want to download (it can be checked
-     by hashing the file). In this case we don't need to download
-     anything. We only need to ensure that `ucDownloadedUpdate` will
-     contain proper value.
+     by hashing the file). In this case we do not download
+     anything. We only need to ensure that `ucDownloadedUpdate` contains the proper value.
    * The hash of the file is not the one we want to download. In this
-     case we delete this file immediately (because we don't need it
+     case we delete this file immediately (because we do not need it
      anyway) and download the update we want.
 2. [Optional] In order to solve the second drawback of the existing
    approach, we can store the hash of the installer we assume to be
-   stored on disk in Misc DB. In this case we can check always check
+   stored on disk in Misc DB. In this case we always check
    the installer's integrity. When we write the installer on disk, we
    first update the hash in Misc DB and only then do
    `BSL.writeFile`. When we want to check the launcher's integrity, we
-   compare its hash with the one in Misc DB. If differs, we delete the
-   installer and assume it doesn't exist.
-3. Impose timeout on downloading function (`downloadUri`). If timeout
+   compare its hash with the one in Misc DB. If it differs, we delete the
+   installer and assume it does not exist.
+3. Impose timeout on downloading the function (`downloadUri`). If a timeout
```

Comment:
Careful on the definite article here in translating from the Russenglish.. It's *the* downloading function. We do not download functions! :-)

Should be:

> Impose **a** timeout on **the** downloading function (`downloadUri` ).

Or perhaps better

> Impose **a** timeout on **the** function `downloadUri`.